### PR TITLE
Feature/add fire base authentication

### PR DESCRIPTION
**Pull request by NathanDVT**
* Improved UI Layout 
* Improved UI test structure
* Added Dashboard screens and dashboard layouts
* Added MVVM Architecture for home screen
Screenshot 2020-03-21 at 22 44 46

### CODE REVIEW

File: realwork/realwork/SignUpBoard/SignUpVC.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import UIKit
+//import Firebase
+import NLibrary
+
+class SignUpVC: UIViewController {
+
+    @IBOutlet weak var edtEmail: UITextField!
+    @IBOutlet weak var edtPassword: UITextField!
+    @IBOutlet weak var loaderIndicator: UIActivityIndicatorView!
+    lazy var userVM: UserVM = {
```
Comment (DavidMinders) [inline realwork/realwork/SignUpBoard/SignUpVC.swift]: I don't think it is necessary to create a lazy var in this way. I think the following should work: 
lazy var userViewModel = UserVM(viewController: self, userRepo UserRepo())

Also, is a lazy init required ? I see that this ViewModel is instantiated again in the viewDidLoad.

**Comment by DavidMinders** [comment]
I have a few suggestions that you can consider:

- You can remove the generated comments (e.g. the comments in the viewDidLoad)
- Remove commented out code before merging into development

**Naming conventions:**
- Many class names use abbreviations (e.g HomeBoardVC). In my opinion it is better to be more descriptive when naming e.g. naming the ViewController something like HomeBoardViewController

- Some of the naming conventions used seem like Java naming conventions. (Java naming conventions vs Swift naming conventions e.g. edtEmail vs emailTextField)

File: realwork/realwork/SignUpBoard/SignUpVC.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    func showMessagePrompt(title: String, message: String) {
+        loaderIndicator.stopAnimating()
+        let alertController = UIAlertController(title: title, message:
+            message, preferredStyle: .alert)
+        alertController.addAction(UIAlertAction(title: "Dismiss", style: .default))
+        self.present(alertController, animated: true, completion: nil)
+    }
+}
+
+extension SignUpVC: SignUpVCprotocol {
```
Comment (DavidMinders) [inline realwork/realwork/SignUpBoard/SignUpVC.swift]: This protocol can also be named SignUpView

Comment (NathanDVT) [reply realwork/realwork/SignUpBoard/SignUpVC.swift]: Is it bad for to desire to keep the word protocol I feel like it is neat, a bit lengthly true but keeps when I do polymorphism, inheritance and such more clearer.

File: realwork/realwork/SignUpBoard/SignUpVC.swift (hunk 1 above)
Comment (NathanDVT) [reply realwork/realwork/SignUpBoard/SignUpVC.swift]: Made the necessary change you required.

**Pull request merged**